src/status_im2/contexts/chat/home/chat_list_item/view.cljs

```diff
@@ -17,61 +19,109 @@
-    {:components (conj (:components acc) (:components result))
-     :length     (+ (:length acc) (:length result))}))
-
-(defn render-subheader
-  "Render the preview of a last message to a maximum of max-subheader-length characters"
+
+(defn parsed-text-to-one-line
   [parsed-text]
-  (let [result
-        (reduce
-         (fn [{:keys [_ length] :as acc-text} new-text-chunk]
-           (if (>= length max-subheader-length)
-             (reduced acc-text)
-             (add-parsed-to-subheader acc-text new-text-chunk)))
-         {:components [quo/text
-                       {:size                :paragraph-2
-                        :style               {:color (colors/theme-colors colors/neutral-50
-                                                                          colors/neutral-40)
-                                              :width "90%"}
-                        :number-of-lines     1
-                        :ellipsize-mode      :tail
-                        :accessibility-label :chat-message-text}]
-          :length     0}
-         parsed-text)]
-    (:components result)))
+  (reduce
+   (fn [acc {:keys [type literal children destination]}]
+     (case type
+       "paragraph"
+       (str acc (parsed-text-to-one-line children) " ")
+
+       "mention"
+       (str acc (rf/sub [:messages/resolve-mention literal]))
+
+       "status-tag"
+       (str acc literal)
+
+       "link"
+       (str acc destination)
+
+       (str acc literal)))
+   ""
+   parsed-text))
+
+(defn extract-text-from-message
+  [{:keys [content]}]
+  (let [{:keys [parsed-text text]} content]
+    (if parsed-text (parsed-text-to-one-line parsed-text) text)))
+
+(defn last-message-preview
+  "Render the preview of a last message to a maximum of max-subheader-length characters"
+  [chat-id {:keys [content-type content outgoing album-images-count] :as message}]
+
+  (rf/dispatch [:pin-message/load-pin-messages chat-id])
+  (let [{:keys [response-to]} content
+        content-text (extract-text-from-message message)
+        preview-text
+        (case content-type
+          constants/content-type-text
+          (if outgoing
+            (str (i18n/label :t/you-colon) " " content-text)
+            content-text)
+
+          constants/content-type-emoji
+          (if outgoing
+            (str (i18n/label :t/you-colon) " " content-text)
+            content-text)
+
+          constants/content-type-system-text
+          (let [pinned-message      (rf/sub [:chats/pinned-message chat-id response-to])
+                pinned-content-text (extract-text-from-message pinned-message)]
+            (if (not (string/blank? pinned-content-text))
+              (if outgoing
+                (str (i18n/label :t/you-pinned) " " pinned-content-text)
+                (str (i18n/label :t/Pinned) " " pinned-content-text))
+              (if outgoing
+                (i18n/label :t/you-pinned-a-message)
+                (i18n/label :t/Pinned-a-message))))
```

Comment:
capital `P`, I think we use lowercase for keys, not sure if we want to change the behavior

Reply:
We already had a few similar translations which are different only in capitalized first letter, so I followed similar pattern , because we already have translation for "pinned messages" but with lowercase "p".